integreat_cms/cms/models/abstract_content_translation.py

```diff
@@ -0,0 +1,427 @@
+    minor_edit = models.BooleanField(
+        default=False,
+        verbose_name=_("minor edit"),
+        help_text=_(
+            "Tick if this change does not require an update of translations in other languages."
+        ),
+    )
+    last_updated = models.DateTimeField(
+        auto_now=True,
+        verbose_name=_("modification date"),
+    )
+    creator = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        verbose_name=_("creator"),
+    )
+    links = GenericRelation(Link, related_query_name="poi_translations")
+
+    @staticmethod
+    def foreign_field():
+        """
+        The field name of the reference to the foreign object which the translation belongs to
+
+        To be implemented in the inheriting model
+        """
+        raise NotImplementedError
+
+    @property
+    def foreign_object(self):
+        """
+        Returns the object the translation belongs to
+        This is needed to generalize the :mod:`~integreat_cms.cms.utils.slug_utils` for all content types
+
+        To be implemented in the inheriting model
+        """
+        raise NotImplementedError
+
+    @property
+    def url_prefix(self):
+        """
+        Generates the prefix of the url of the content translation object
+
+        For information about the components of such an url,
+        see :meth:`~integreat_cms.cms.models.abstract_content_translation.AbstractContentTranslation.get_absolute_url`
+
+        :return: The prefix to the url
+        :rtype: str
+        """
+        return (
+            "/"
+            + "/".join(
+                filter(
+                    None,
+                    [
+                        self.foreign_object.region.slug,
+                        self.language.slug,
+                        self.url_infix,
+                    ],
+                )
+            )
+            + "/"
+        )
+
+    @property
+    def url_infix(self):
+        """
+        Generates the infix of the url of the content translation object
+
+        For information about the components of such an url,
+        see :meth:`~integreat_cms.cms.models.abstract_content_translation.AbstractContentTranslation.get_absolute_url`
+
+        To be implemented in the inheriting model
+        """
+        raise NotImplementedError
+
+    def get_absolute_url(self):
+        """
+        Generates the absolute url of the content translation object
+
+        Here is an example for demonstrating the components of a page url::
+
+            https://integreat.app/augsburg/en/welcome/city-map/attractions/
+                                 ------------------------------------------ absolute url
+                                 ------------------------------             url prefix
+                                              ----------------              url infix
+                                                               -----------  slug
```

Comment:
```suggestion
                                 |----------------------------------------|    get_absolute_url()
            |-------------------------------------------------|                base_link
                                 |----------------------------|                url_prefix
                                             |----------------|                url_infix
                                                              |-----------|    slug
```
I love this breakdown and I think it really helps understanding the structure, I'd just add `base_link` to the list and add a few delimiters.